### DISCUSSION TITLE
Keyboard navigation in search results

### DISCUSSION THREAD

**Discussion by illright** [withastro/starlight#960] [Feature Requests]
### What version of `starlight` are you using?

0.10.2

### What is your idea?

I'm used to being able to use Up and Down arrow keys to navigate through search results, I'd like to see this added to Starlight's search.

### Why is this feature necessary?

It would help with accessibilty for users that don't use the mouse and want to navigate quicker than the regular Tab order. It would also unify the search experience with Algolia's search, which is a popular option in Docusaurus-backed docs.

### Do you have examples of this feature in other projects?

- https://docusaurus.io/ has that on their home page

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by HiDeoo** [comment]
To add a little bit more context, this is something that has been brought up [before](https://discord.com/channels/830184174198718474/1070481941863878697/1124026439554052137) on Discord and it [looks like](https://discord.com/channels/830184174198718474/1070481941863878697/1149827542702899251) the focus is going to shift from Pagefind’s Default UI (that Starlight is currently using) to their Modular UI soon and the [plan for Starlight](https://discord.com/channels/830184174198718474/1070481941863878697/1151645904273682473) may be to switch to the Modular UI at some point which may also be a better place/time to implement this feature.

Nonetheless, this is definitely a good issue to track this feature, thanks for bringing it up! 👍

**Comment by delucis** [comment]
I’ll add here that this is might also be something best improved in Pagefind itself as I’m guessing its utility extends beyond Starlight and to other Pagefind users.

**Reply by jermanuts** [reply]
hey @bglw is this planned? ❤️

**Comment by mayank99** [comment]
Is there an easy way to implement this in user-land? I'm guessing I would have to [override](https://starlight.astro.build/guides/overriding-components/) the entire [Search](https://github.com/withastro/starlight/blob/main/packages/starlight/components/Search.astro) component.

I don't mind writing the code to build a [combobox](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/), but I would love to be able to reuse the styles that have already been written by Starlight.

**Reply by delucis** [reply]
I think it's a bit tricky to customise the behaviour directly because the DOM for the search component gets generated by Pagefind's compiled Svelte components (which we then apply some additional CSS on top of). 

But if you wanted to use only the styles and write the combobox logic yourself, I think that might be possible? We import Pagefind's styles ourselves, so you could import those. And I guess we could export our own Search styles so they could be reused?

That said, if you're interested, we've always discussed building our own search UI from scratch or adding these improvements upstream to Pagefind itself, so there may also be scope for improving things for all users.
